src/com/palmergames/bukkit/towny/utils/AreaSelectionUtil.java

```diff
@@ -72,101 +86,178 @@
-				} else if (owner instanceof Resident) {
-					available = TownySettings.getMaxResidentPlots((Resident) owner);
-				}
-				
+		if (args.length > 0) {
+			int r = MAX_RECT_RADIUS;  // The radius of the claim.
+
+			/*
+			 *  Area selections are capped at a 15 radius which should be a 31x31 (or a square with a side of 496 blocks in length.)  
+			 *  Players need a permission node to use area claims and the max radius usable defaults to 4 (set in the config.)
+			 */
+
+			if (args[0].equalsIgnoreCase("auto")) {
 				
+				/*
+				 * Select everything possible in a rectangle shape.
+				 */
 
-				if (args[0].equalsIgnoreCase("auto")) {
-					// Attempt to select outwards until no town blocks remain
+				if (TownySettings.getMaxClaimRadiusValue() > 0) 
+					r = Math.min(r, TownySettings.getMaxClaimRadiusValue());
 
-					while (available - Math.pow((r + 1) * 2 - 1, 2) >= 0)
-						r += 1;
+			} else {
+			
+				/*
+				 * Select an area that will claim a perfect square shape, using a given radius
+				 * or the reduced radius that will give a perfect square.
+				 */
+				
+				try {
+					r = Integer.parseInt(args[0]);
+				} catch (NumberFormatException e) {
+					throw new TownyException(Translation.of("msg_err_invalid_radius"));
+				}
+				if (TownySettings.getMaxClaimRadiusValue() > 0 && r > TownySettings.getMaxClaimRadiusValue())
+					throw new TownyException(Translation.of("msg_err_invalid_radius_number", TownySettings.getMaxClaimRadiusValue()));
 
-				} else {
-					try {
-						r = Integer.parseInt(args[0]);
-					} catch (NumberFormatException e) {
-						throw new TownyException(Translation.of("msg_err_invalid_radius"));
+				// Calculate how many TownBlocks are needed to claim the radius. 
+				// Start blocks at 0 if the plot is unclaimed.
+				int neededBlocks = pos.getTownBlock().hasTown() ? 0 : 1;
+				neededBlocks += 4 * r * (r + 1);
+				
+				// Rethink how much of a radius will be used, as there's not enough available TownBlocks.
+				if (neededBlocks > available) {
+					r = 1; 
+					int total = 0;
+					while (available - ((r * 8) + total) >= 0) { // Radius 1 grabs 8 blocks, 2 grabs 16 more requiring 24, 3 grabs 24 more requiring 48, etc...
+						total += r * 8;
+						r++;
 					}
+					r--; // Finally reduce the radius by 1 (so that we have a perfect ring of claims,) and replace the original r.
 				}
-				if (r > TownySettings.getMaxClaimRadiusValue() && TownySettings.getMaxClaimRadiusValue() > 0) {
-					throw new TownyException(Translation.of("msg_err_invalid_radius_number", TownySettings.getMaxClaimRadiusValue()));
+			}
+
+			/*
+			 * Adds WorldCoords in a spiral-out pattern.
+			 */
+			int sideLength = (r * 2) + 1;
+			int x = 0, z = 0, dx = 0, dz = -1;
+			int t = r;
+			for (int i = 0; i <= available; i++) {
+				if ((-sideLength / 2 <= x) && (x <= sideLength / 2) && (-sideLength / 2 <= z) && (z <= sideLength / 2)) {
+					out.add(new WorldCoord(pos.getWorldName(), pos.add(x, z)));
```

Comment:
`pos.add()` returns a new worldcoord, so this entire line can be replaced with `out.add(pos.add(x,z));`